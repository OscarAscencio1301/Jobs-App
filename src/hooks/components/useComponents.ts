import { useDispatch, useSelector } from "react-redux"
import { GlobalSlice } from "../../interfaces/global/global"
import { Componentes } from "../../interfaces/components/components"
import { changeAlert, changeModal } from "../../store/components/components.Slice"


export const useComponents = () => {

    const { isOpenAlert, isOpenModal } = useSelector<GlobalSlice, Componentes>(state => state.components)
    const dispatch = useDispatch()

    const changeViewModal = () => {
        dispatch(changeModal())
    }

    const changeViewAlert = () => {
        dispatch(changeAlert())
    }

    return {
        isOpenModal,
        isOpenAlert,
        changeViewModal,
        changeViewAlert
    }
}
