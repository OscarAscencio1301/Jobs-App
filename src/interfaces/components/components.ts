export interface Componentes {
    isOpenModal: boolean,
    isOpenAlert: boolean,
}