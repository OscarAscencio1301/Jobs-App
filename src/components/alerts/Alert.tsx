
import { FcApproval } from "react-icons/fc";
import { useComponents } from "../../hooks/components/useComponents";

export const Alert = ({ text = "Cambios aplicados" }: { text?: string }) => {

  const {isOpenAlert, changeViewAlert} = useComponents()

  return (
    <div className={`w-full h-full bg-opacity-50 bg-black fixed top-0 left-0 flex items-center justify-center z-50 ${isOpenAlert ? 'flex' : 'hidden'}`}>
      <div className="bg-white flex flex-col items-center justify-center gap-4 p-5 rounded-md">
        <FcApproval size="120px" />
        <h2 className="text-3xl">{text}</h2>
        <button className="bg-green-600 text-white py-2 px-10 rounded-md flex-1" onClick={changeViewAlert}>Aceptar</button>
      </div>
    </div>
  )
}
