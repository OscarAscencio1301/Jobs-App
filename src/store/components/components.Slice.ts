import { createSlice } from "@reduxjs/toolkit";
import { Componentes } from "../../interfaces/components/components";

const initialState: Componentes = {
    isOpenModal: false
}


export const componentsSlice = createSlice({
    name: 'components',
    initialState,
    reducers: {
        changeModal: (state) => {
            state.isOpenModal = !state.isOpenModal
        }
    }
})

export const { changeModal } = componentsSlice.actions
