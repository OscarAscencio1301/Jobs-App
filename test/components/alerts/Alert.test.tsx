import { fireEvent, render, screen } from '@testing-library/react'
import { Alert } from '../../../src/components/alerts/Alert';
import { useComponents } from '../../../src/hooks/components/useComponents';

jest.mock('../../../src/hooks/components/useComponents')

describe('Testing Alert', () => {
    const changeViewAlertMock = jest.fn()

    beforeEach(() => jest.clearAllMocks())

    test('Initial State', () => {

        (useComponents as jest.Mock).mockReturnValue({
            isOpenModal: false,
            isOpenAlert: false,
            changeViewModal: jest.fn(),
            changeViewAlert: changeViewAlertMock,
        })

        const { container } = render(<Alert />)
        expect(container).toMatchSnapshot()
    });

    test('Initial Text', () => {

        (useComponents as jest.Mock).mockReturnValue({
            isOpenModal: false,
            isOpenAlert: true,
            changeViewModal: jest.fn(),
            changeViewAlert: changeViewAlertMock,
        })

        render(<Alert />)

        expect(screen.getByText('Cambios aplicados')).toBeTruthy()
    });

    test('Close Alert', () => {

        (useComponents as jest.Mock).mockReturnValue({
            isOpenModal: false,
            isOpenAlert: true,
            changeViewModal: jest.fn(),
            changeViewAlert: changeViewAlertMock,
        })

        render(<Alert />)
        const button = screen.getByRole('button')
        fireEvent.click(button)

        expect(changeViewAlertMock).toBeCalled()
    });
});