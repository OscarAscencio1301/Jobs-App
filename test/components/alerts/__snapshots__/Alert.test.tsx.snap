// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Alert Initial State 1`] = `
<div>
  <div
    class="w-full h-full bg-opacity-50 bg-black fixed top-0 left-0 flex items-center justify-center z-50 hidden"
  >
    <div
      class="bg-white flex flex-col items-center justify-center gap-4 p-5 rounded-md"
    >
      <svg
        enable-background="new 0 0 48 48"
        fill="currentColor"
        height="120px"
        stroke="currentColor"
        stroke-width="0"
        version="1"
        viewBox="0 0 48 48"
        width="120px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polygon
          fill="#8BC34A"
          points="24,3 28.7,6.6 34.5,5.8 36.7,11.3 42.2,13.5 41.4,19.3 45,24 41.4,28.7 42.2,34.5 36.7,36.7 34.5,42.2 28.7,41.4 24,45 19.3,41.4 13.5,42.2 11.3,36.7 5.8,34.5 6.6,28.7 3,24 6.6,19.3 5.8,13.5 11.3,11.3 13.5,5.8 19.3,6.6"
        />
        <polygon
          fill="#CCFF90"
          points="34.6,14.6 21,28.2 15.4,22.6 12.6,25.4 21,33.8 37.4,17.4"
        />
      </svg>
      <h2
        class="text-3xl"
      >
        Cambios aplicados
      </h2>
      <button
        class="bg-green-600 text-white py-2 px-10 rounded-md flex-1"
      >
        Aceptar
      </button>
    </div>
  </div>
</div>
`;
